File: Backend/Backend/Backend/Business/src/Utils/Response.cs
namespace Backend.Business.src.Utils
{
    public class Response<T>
    {
        public T value { get; set; }
        public bool exceptionHasOccured { get; set; }
        public string errorMessage { get; set; }
        
        
        public Response(T value)
        {
            this.value = value;
            this.exceptionHasOccured = false;
            this.errorMessage = "";
        }
        
        public Response(bool exceptionHasOccured, string errorMessage)
        {
            this.exceptionHasOccured = exceptionHasOccured;
            this.errorMessage = errorMessage;
        }
    }
}

File: Frontend/src/services/Response.js
export class Response {
    constructor(response) {
        this.value = response.value;
        this.exceptionHasOccured = response.exceptionHasOccured;
        this.errorMessage=response.errorMessage
    }
    static create(value, exceptionHasOccured,errorMessage) {
        const res= new Response({
            value: value,
            exceptionHasOccured: exceptionHasOccured,
            errorMessage:errorMessage
        })
        return res

    }
}

